NationStatesAPIBot/NationStatesAPIBot/Services/DumpDataService.cs

```diff
@@ -140,69 +142,118 @@ private async Task InitialUpdate()
 
         private void LoadDumpsFromStream(GZipStream regionsStream, GZipStream nationsStream)
         {
-            _regions = GetRegionsFromCompressedStream(regionsStream);
-            _nations = GetNationsFromCompressedStream(nationsStream);
-            AddNationsToRegions();
+            Stopwatch stopWatch = Stopwatch.StartNew();
+            _regions = GetRegionsFromStream(regionsStream);
+            stopWatch.Stop();
+            _logger.LogDebug(defaultEventId, GetLogMessage($"Parsing regions took {stopWatch.Elapsed} to complete."));
+            stopWatch.Restart();
+            _nations = GetNationsFromStream(nationsStream);
+            stopWatch.Stop();
+            _logger.LogDebug(defaultEventId, GetLogMessage($"Parsing nations took {stopWatch.Elapsed} to complete."));
             DataAvailable = true;
         }
 
         private void ReadDumpsFromLocalFileSystem()
         {
-            GZipStream regionStream;
-            GZipStream nationStream;
-            using (var fs = new FileStream(regionFileName, FileMode.Open, FileAccess.Read, FileShare.Read))
+            GZipStream regionStream = null;
+            GZipStream nationStream = null;
+            FileStream fsr = null;
+            FileStream fsn = null;
+            try
             {
-                regionStream = new GZipStream(fs, CompressionMode.Decompress);
+                Stopwatch stopWatch = Stopwatch.StartNew();
+                fsr = new FileStream(regionFileName, FileMode.Open, FileAccess.Read, FileShare.Read);
+                regionStream = new GZipStream(fsr, CompressionMode.Decompress);
+
+                stopWatch.Stop();
+                _logger.LogDebug(defaultEventId, GetLogMessage($"Reading region dump from local cache took {stopWatch.Elapsed} to complete."));
+                stopWatch.Restart();
+                fsn = new FileStream(nationFileName, FileMode.Open, FileAccess.Read, FileShare.Read);
+                nationStream = new GZipStream(fsn, CompressionMode.Decompress);
+
+                stopWatch.Stop();
+                _logger.LogDebug(defaultEventId, GetLogMessage($"Reading nation dump from local cache took {stopWatch.Elapsed} to complete."));
+                var fileInfoNations = new FileInfo(nationFileName);
+                lastDumpUpdateTime = fileInfoNations.CreationTimeUtc;
+                LoadDumpsFromStream(regionStream, nationStream);
             }
-            using (var fs = new FileStream(nationFileName, FileMode.Open, FileAccess.Read, FileShare.Read))
+            finally
             {
-                nationStream = new GZipStream(fs, CompressionMode.Decompress);
+                fsr?.Close();
+                fsn?.Close();
+                nationStream?.Close();
+                regionStream?.Close();
+                fsr.Close();
+                fsn.Close();
+                nationStream?.Dispose();
```

Comment:
Why do you Close and Dispose? Are they both needed?

Reply:
No they don't, just to be sure that everything is probably closed and disposed after usage.